crates/matrix-sdk/src/widget_api/handler/mod.rs

```diff
@@ -52,89 +55,55 @@ impl<T: Driver> MessageHandler<T> {
             }
 
             Incoming::GetOpenID(r) => {
-                let state = self.driver.get_openid(r.clone()).await;
+                let state = self.client.get_openid(r.clone()).await;
                 r.reply(Ok((&state).into()))?;
 
                 if let OpenIDState::Pending(resolution) = state {
                     let resolved = resolution.await.map_err(|_| Error::WidgetDied)?;
-                    let (req, resp) = Request::new(resolved.into());
-                    self.driver.send(Outgoing::OpenIDUpdated(req)).await?;
-                    resp.recv().await?;
+                    self.widget.send(outgoing::OpenIDUpdated(resolved.into())).await?;
                 }
             }
 
             Incoming::ReadEvents(r) => {
-                let response = self.read_events(&r).await;
+                let response = self.read_events(&*r).await;
                 r.reply(response)?;
             }
 
             Incoming::SendEvent(r) => {
-                let response = self.send_event(&r).await;
+                let response = self.send_event(&*r).await;
                 r.reply(response)?;
             }
```

Comment:
By inlining their source code? (I'm not against this idea) Or did you mean something else?

Reply:
yea inlining the source of the sen/read_event fn's

Reply:
This is a good point and I agree with it generally, but I noticed that there is one reason why it's currently separate - error handling. We use `?` inside it to handle all the errors, so that in the main handler function we convert the error into the response by calling `reply(response)` where the response is a `Result`.